cloud/amazon/s3_website.py

```diff
@@ -0,0 +1,291 @@
+        website_configuration['IndexDocument'] = { 'Suffix': suffix }
+
+    if redirect_all_requests is not None:
+        website_configuration['RedirectAllRequestsTo'] = _create_redirect_dict(redirect_all_requests)
+
+    return website_configuration
+
+
+def enable_or_update_bucket_as_website(client_connection, resource_connection, module):
+    
+    bucket_name = module.params.get("name")
+    redirect_all_requests = module.params.get("redirect_all_requests")
+    # If redirect_all_requests is set then don't use the default suffix that has been set
+    if redirect_all_requests is not None:
+        suffix = None
+    else:
+        suffix = module.params.get("suffix")
+    error_key = module.params.get("error_key")
+    changed = False
+
+    try:
+        bucket_website = resource_connection.BucketWebsite(bucket_name)
+    except ClientError as e:
+        module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    try:
+        website_config = client_connection.get_bucket_website(Bucket=bucket_name)
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'NoSuchWebsiteConfiguration':
+            website_config = None
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    if website_config is None:
+        try:
+            bucket_website.put(WebsiteConfiguration=_create_website_configuration(suffix, error_key, redirect_all_requests))
+            changed = True
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+        except ValueError as e:
+            module.fail_json(msg=str(e))
+    else:
+        try:
+            if (suffix is not None and website_config['IndexDocument']['Suffix'] != suffix) or \
+                    (error_key is not None and website_config['ErrorDocument']['Key'] != error_key) or \
+                    (redirect_all_requests is not None and website_config['RedirectAllRequestsTo'] != _create_redirect_dict(redirect_all_requests)):
+
+                try:
+                    bucket_website.put(WebsiteConfiguration=_create_website_configuration(suffix, error_key, redirect_all_requests))
+                    changed = True
+                except (ClientError, ParamValidationError) as e:
+                    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+        except KeyError as e:
+            try:
+                bucket_website.put(WebsiteConfiguration=_create_website_configuration(suffix, error_key, redirect_all_requests))
+                changed = True
+            except (ClientError, ParamValidationError) as e:
+                module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+        except ValueError as e:
+            module.fail_json(msg=str(e))
+
+        # Wait 5 secs before getting the website_config again to give it time to update
+        sleep(5)
```

Comment:
Use a regular import of system python modules instead of using the versions imported from basic.py:*
ie,
``` python
import time
<...>
                      time.sleep(5)
```